src/test/java/part2/exercise/CollectorsExercise1.java

```diff
@@ -58,102 +63,182 @@ public int getDuration() {
+                           employee -> employee.getValue().getPerson()));
+
+        //third option
+        Map<String, Person> option3 = persons
+            .collect(new Collector<PersonPositionDuration, Map<String, PersonPositionDuration>, Map<String, Person>>() {
+                @Override
+                public Supplier<Map<String, PersonPositionDuration>> supplier() {
+                    return HashMap::new;
+                }
+
+                @Override
+                public BiConsumer<Map<String, PersonPositionDuration>, PersonPositionDuration> accumulator() {
+                    return (map, person) -> map.merge(
+                        person.getPosition(),
+                        person,
+                        (ppd1, ppd2) -> ppd1.getDuration() > ppd2.getDuration() ? ppd1 : ppd2);
+                }
+
+                @Override
+                public BinaryOperator<Map<String, PersonPositionDuration>> combiner() {
+                    return (map1, map2) -> {
+                        map2.forEach((key, value) -> map1.merge(key, value, (ppd1, ppd2) ->
+                            ppd1.getDuration() > ppd2.getDuration() ? ppd1 : ppd2));
+                        return map1;
+                    };
+                }
+
+                @Override
+                public Function<Map<String, PersonPositionDuration>, Map<String, Person>> finisher() {
+                    return (map) -> map.entrySet().stream()
+                        .collect(toMap(Map.Entry::getKey,
+                                       employee -> employee.getValue().getPerson()));
+                }
+
+                @Override
+                public Set<Characteristics> characteristics() {
+                    return Collections.emptySet();
+                }
+            });
+
+        return option1;
     }
 
     @Test
     public void getTheCoolestOne2() {
         final Map<String, Person> coolestByPosition = getCoolestByPosition2(getEmployees());
 
-        coolestByPosition.forEach((position, person) -> System.out.println(position + " -> " + person));
+        assertEquals(new Person("John", "Doe", 30), coolestByPosition.get("QA"));
+        assertEquals(new Person("John", "Galt", 26), coolestByPosition.get("dev"));
+        assertEquals(new Person("John", "Doe", 24), coolestByPosition.get("BA"));
     }
 
-    // With the longest sum duration on this position
-    // { John Doe, [{dev, google, 4}, {dev, epam, 4}] } предпочтительнее, чем { A B, [{dev, google, 6}, {QA, epam, 100}]}
+//     With the longest sum duration on this position
+//     { John Doe, [{dev, google, 4}, {dev, epam, 4}] } предпочтительнее, чем { A B, [{dev, google, 6}, {QA, epam, 100}]}
     private Map<String, Person> getCoolestByPosition2(List<Employee> employees) {
-        // TODO
-        throw new UnsupportedOperationException();
+        Map<String, List<PersonPositionDuration>> map = employees.stream()
+            .flatMap(this::positionDuration)

```

Comment:
Simplify. You can ad additional method to aggregate durations for a single person like this `flatMap(this::groupPersonDurations)`